### PR TITLE
Added Triton's config route

### LINKED ISSUE #3648: Add support for /v2/models/{model name}/config endpoint in Executor
This is part of Triton extensions https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_model_configuration.md

### PR DESCRIPTION
**Pull request by naor2013**
**What this PR does / why we need it**:
This PR adds the /v2/models/{model}/config route that Triton supports.

**Which issue(s) this PR fixes**:

Fixes #3648 

**Special notes for your reviewer**:
I do not know how to run tests or add new tests.
I looked into existing tests in similar places, they all use the v1 endpoint which is not relevant here.
If v2 endpoints aren't even tested anyway, can this be approved without new tests as well?
If not, can a maintainer help me add them?

**Does this PR introduce a user-facing change?**:

```release-note
Added Triton's config route
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @naor2013. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **gsunner**
You can assign the PR to them by writing `/assign @gsunner` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by naor2013** [comment]
@gsunner @cliveseldon @axsaucedo Any chance one of you helps me with this pull request? All the tests seem to pass except for the 2 that seem to fail in all PRs.

File: executor/api/rest/constants.go
Comment (cliveseldon) [inline executor/api/rest/constants.go:7]: Typo - "condig"

**Review by adriangonz — CHANGES_REQUESTED** [review]
Nice one  @naor2013! :rocket: 

I've just added a small comment below. 

Besides that, could you add some tests, either at the "operator" or "integration" level (or both)? For the former, it would be enough to mock the `/config` HTTP endpoint to ensure it gets called. Whereas, for the latter, you could write a test spec that deploys a Triton `SeldonDeployment` manifest and then hit the actual `/config` endpoint to ensure the executor forwards the request (e.g. see the [`test_sklearn_v2`](https://github.com/adriangonz/seldon-core/blob/b688218c115939413c6a9a17b54f433d4ad1e4b0/testing/scripts/test_prepackaged_servers.py#L55-L83)).

**Comment by naor2013** [comment]
@adriangonz I looked into the test example, it looks like it's a sklearn model that it takes out of your google storage. Do you have a triton example in your google storage I can use?

**Comment by cliveseldon** [comment]
You can find some [here](https://github.com/SeldonIO/seldon-core/blob/master/notebooks/triton_examples.ipynb)

**Comment by naor2013** [comment]
@cliveseldon @adriangonz Can you guys help me understand how to implement it for grpc as well? I don't really understand the .proto and .pb.go and other files there.
Do you think you guys can add it yourselves soon after I finish my PR (before the new version) or should I spend some more time with your help to figure it out?

**Comment by cliveseldon** [comment]
@naor2013 There is already a [stub](https://github.com/SeldonIO/seldon-core/blob/5b327375a9cb43b7806d18c2a1eee83d9490b2d8/executor/api/grpc/kfserving/server.go#L94-L96) which you would need to flesh out. Hopefully the rest should be straightforward  to tie in with the same method you used for REST?

**Comment by axsaucedo** [comment]
@naor2013 just following up with the discussion on this PR - did you have a chance to explore some of the points above?

**Comment by axsaucedo** [comment]
@naor2013 just following up on this, looking to see if this is something that we'd be able to work into our next release (within the next week or two), or if you no longer have time we can park it for now and look at this again in the near future?

**Comment by axsaucedo** [comment]
@naor2013 thank you again for the PR, so far seems like great progress - we'll close the PR due to inactivity but if you get a chance to have a look in the near future, we'd be keen to pick it up again

**Pull request closed without merging**
